src/revocation.rs

```diff
@@ -8,9 +8,99 @@ use crate::error::{Error, Result};
 use crate::secure_mount;
 
 use std::path::Path;
+use std::process::Command;
 
 use serde_json::Value;
 
+/// Runs a python script (used for revocation actions)
+pub(crate) fn run_python(dir: &Path, script: &str) -> Result<()> {
+    let status =
+        // TODO: Should we assume that these script names are all correctly 
+        // formatted and end with .py? Or should we allow some wiggle room and
+        // append .py if it isn't there already?
+        match Command::new("python").current_dir(dir).arg(script).status() {
+            Ok(status) => status,
+            Err(e) => {
+                let msg = format!("ERROR: failed to run revocation action: {:?}", script);
+                // TODO: Unless errors are already logged automatically?
+                error!("{}", msg);
+                return Err(Error::Other(msg));
+            }
+        };
+
+    if !status.success() {
+        let msg = format!(
+            "ERROR: revocation action {:?} returned with status {:?}",
+            script, status
+        );
+        error!("{}", msg);
+        return Err(Error::Other(msg));
+    }
+
+    info!(
+        "{}",
+        format!("INFO: revocation action {:?} successful", script)
+    );
+    Ok(())
+}
+
+/// Runs revocation actions
+// TODO: Do we actually need this msg for anything?
+pub(crate) fn run_revocation_actions(_msg: Value) -> Result<()> {
+    pretty_env_logger::init();
+
+    // Check Python version > 3.x
+    // TODO: does returning an error here make sure the agent is in a bad /
+    // failed security state until it is able to execute these scripts successfully?
+    // otherwise this could be an attack vector
+    let python_version = String::from_utf8(
+        Command::new("python")
+            .arg("-V")
+            .output()
+            .expect("could not find Python in $PATH")
+            .stdout,
+    )?;
+
+    if python_version.split_whitespace().collect::<Vec<&str>>()[1]
```

Comment:
nit: It would be ideal to have this occur once when the agent initializes.

Reply:
Great idea - this takes care of my worry about the scripts not executing, also. I've moved it to its own function `check_system_prerequisites()` in `main.rs`.